File: shared/types/regime.ts
// Mirrors backend/api/schemas/regime.py

export type RegimeLabelValue =
  | "low_vol"
  | "medium_vol"
  | "high_vol"
  | "high_vol_jumps"
  | "trending"
  | "sideways"
  | "stress_crisis";

export interface RegimeResponse {
  ticker: string;
  as_of: string;
  label: RegimeLabelValue;
  confidence: number;
  posterior: Record<string, number>;
  raw_state_id: number;
  method: "hmm" | "gmm";
  n_states_fit: number;
}

export interface RegimeTimelinePoint {
  date: string;
  label: RegimeLabelValue;
  confidence: number;
}

export interface RegimeTimelineResponse {
  ticker: string;
  points: RegimeTimelinePoint[];
}

File: backend/api/schemas/regime.py
from __future__ import annotations

from datetime import date

from pydantic import BaseModel


class RegimeResponse(BaseModel):
    ticker: str
    as_of: date
    label: str
    confidence: float
    posterior: dict[str, float]
    raw_state_id: int
    method: str
    n_states_fit: int


class RegimeTimelinePoint(BaseModel):
    date: date
    label: str
    confidence: float


class RegimeTimelineResponse(BaseModel):
    ticker: str
    points: list[RegimeTimelinePoint]
